src/Components/WebAssembly/WebAssembly/src/Rendering/WebAssemblyRenderer.cs

```diff
@@ -22,16 +22,21 @@ namespace Microsoft.AspNetCore.Components.WebAssembly.Rendering;
 internal sealed partial class WebAssemblyRenderer : WebRenderer
 {
     private readonly ILogger _logger;
+    private readonly Dispatcher _dispatcher;
 
     public WebAssemblyRenderer(IServiceProvider serviceProvider, ILoggerFactory loggerFactory, JSComponentInterop jsComponentInterop)
         : base(serviceProvider, loggerFactory, DefaultWebAssemblyJSRuntime.Instance.ReadJsonSerializerOptions(), jsComponentInterop)
     {
         _logger = loggerFactory.CreateLogger<WebAssemblyRenderer>();
+        // if SynchronizationContext.Current is not JSSynchronizationContext on main thread, we are in single-threaded flavor of the dotnet wasm runtime or on the server side.
+        _dispatcher = SynchronizationContext.Current == null || SynchronizationContext.Current.GetType().FullName != "System.Runtime.InteropServices.JavaScript.JSSynchronizationContext"
```

Comment:
>  SynchronizationContext.Current.GetType().FullName != "System.Runtime.InteropServices.JavaScript.JSSynchronizationContext"

I don't think we need this check, could we avoid it? The Webassembly renderer only ever runs in Webassembly, and I'd rather avoid taking a dependency on the type name here.

Reply:
3rd party code could install any `SynchronizationContext`.

Reply:
> 3rd party code could install any SynchronizationContext.

I don't think we permit this in other Blazor scenarios. That is, you can use your own sync context from your own code if you want, but Blazor's `Renderer` is going to use its own sync context no matter what, and that's the sync context you have to dispatch to when you interact with the renderer.

Having a similar restriction here would be fine. Desirable, even.

Reply:
So, this could rather become assert